apps/speed.c

```diff
@@ -3161,6 +3559,365 @@ int speed_main(int argc, char **argv)
         }
     }
 #endif  /* OPENSSL_NO_DH */
+
+    for (testnum = 0; testnum < kems_algs_len; testnum++) {
+        int kem_checks = 1;
+        const char *kem_name = kems_algname[testnum];
+
+        if (!kems_doit[testnum] || !do_kems)
+            continue;
+
+        for (i = 0; i < loopargs_len; i++) {
+            EVP_PKEY *pkey = NULL;
+            EVP_PKEY_CTX *kem_gen_ctx = NULL;
+            EVP_PKEY_CTX *kem_encaps_ctx = NULL;
+            EVP_PKEY_CTX *kem_decaps_ctx = NULL;
+            size_t send_secret_len, out_len;
+            size_t rcv_secret_len;
+            unsigned char *out = NULL, *send_secret = NULL, *rcv_secret;
+            size_t bits;
+            char *name;
+            OSSL_PARAM params[] = { OSSL_PARAM_END, OSSL_PARAM_END };
+            int use_params = 0;
+
+            if (ERR_peek_error()) {
+                BIO_printf(bio_err,
+                           "WARNING: the error queue contains previous unhandled errors.\n");
+                ERR_print_errors(bio_err);
+            }
+
+            if (strncmp(kem_name, "rsa", 3) == 0) {
```

Comment:
I'm counting four `strncmp(kem_name, "rsa", 3)...`  in this code.

Reply:
This is intentional. The alternative would be to introduce new variables (for all classic crypto KEMs basically) and use them throughout. IMO this would make the code less readable. If this is preferred, I'd do this of course. Something along the lines of  `int is_rsa_kem, is_ec_kem, is_x25519_kem, is_x448_kem, is_dhkem`. Please let me know whether you indeed prefer this, @FdaSilvaYY.